tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -101,21 +101,41 @@ int RelayHandler::open(const ACE_INET_Addr& local)
 int RelayHandler::handle_input(ACE_HANDLE)
 {
   ACE_INET_Addr remote;
-  iovec iov;
-  const auto bytes = socket_.recv(&iov, remote);
+  int inlen = 65536; // Default to maximum datagram size.
+  const int iov_size = 1;
+  iovec iov[iov_size];
+  OpenDDS::DCPS::Message_Block_Shared_Ptr buffer;
```

Comment:
Code may be clearer if this was pushed down to where it's used.  I think it would be simpler if the `inlen < 0` case was taken care of first